core/src/main/scala/dagr/core/tasksystem/Task.scala

```diff
@@ -154,6 +154,9 @@ trait Task extends Dependable {
   // A string set by the enclosing workflow to distinguish the task
   private[tasksystem] var contextName : Option[String] = None
 
+  /** The unique id given to this task by the execution system, or None if not being tracked. */
+  private[core] var id : Option[BigInt] = None
```

Comment:
Just wondering if this should be `_id` or `taskId`, so as not to prevent concrete task classes from defining their own property called `id` in the future?